docs/DeveloperGuide.md

```diff
@@ -1,34 +1,663 @@
+
+* Sets up the text editor
+* Loads existing file from Cheatsheet directory within a module for the edit cheat-sheet command
+* Flushes out the text from the editor when a different or new file is opened.
+* Adjusts the font size of the text within the editor
+* Detects mouse input to change font style and save the text
+* Saves the text from the text editor into a file
+
+#### ShortcutListener
+
+* Detects keyboard input for shortcuts
+
+&nbsp;
+
+### Common classes
+
+Classes that are used by multiple components:
+* `CommonMethods`: Stores methods that are used by multiple components
+* `Constants`: Stores constants
+* `Messages`: Stores strings that are printed by the `UI`
+* `DashboardCommands`: Enum of commands that can be used outside a module
+* `ModuleCommands`: Enum of commands that can be used inside a module
+
+&nbsp;
+
+[🡅 Back to Table of Contents](#table-of-contents)
+
+----
+
+## Implementation
+
+In this section, we highlight a few of the key features whose implementations are reflective of most of the commands available, as well as those that are more unique to GULIO.
+
+### Add Lesson
+
+The `AddLessonCommand` class is responsible for the creation and addition of a new `Lesson` object to the lesson list of a given module. The following sequence diagrams shows how a new `Lesson` is created and added to the lesson list.
+
+<p align="center">
+    <img width="973" src="developerGuideImages/addLesson1.png" alt="parse() Sequence Diagram"><br>
+    Figure 7 - parse() Sequence Diagram
+</p>
+
+The creation process is facilitated by the `Parser` class, which parses the appropriate arguments from the user input and initialises the `Lesson` object attributes with the parsed values.
+
+<p align="center">
+    <img width="973" src="developerGuideImages/addLessonCommand.png" alt="AddLessonCommand Constructor Sequence Diagram"><br>
+    Figure 8 - AddLessonCommand Constructor Sequence Diagram
+</p>
+
+The newly created `Lesson` object is then passed to a new `AddLessonCommand` object as an argument.
+
+<p align="center">
+    <img width="973" src="developerGuideImages/add_lesson_to_lesson_list.png" alt="execute() AddLessonCommand Sequence Diagram"><br>
+    Figure 9 - execute() AddLessonCommand Sequence Diagram
+</p>
+
+`AddLessonCommand` then adds the `Lesson` object to the lesson list of a module. The lessons in the list are sorted by their lesson types each time a new lesson is added. `AddLessonCommand` also calls the `writeLesson()` method of `ModuleList` to update the change locally.
+
+&nbsp;
+
+### Adding of Cheat-Sheet
+
+The `AddCheatSheetCommand` class enables the creation, addition and saving of a ".txt" file to the current module’s “Cheatsheet” directory (see Figure 5). Upon creating a new instance of it and calling the `execute()` method on it, the GULIO Text Editor application will also be automatically started.
+
+An invocation of the `add cheat-sheet` command involves the following interactions:
+
+<p align="center">
```

Comment:
The text in the sequence diagram could be too small to read. Maybe you can adjust the size of the diagram to make it more readable?